tools/filter_cells_and_genes/filter_cells_and_genes.xml

```diff
@@ -0,0 +1,246 @@
+<tool id="filter_cells_and_genes" name="Filter out cells and genes" version="0.1.0">
+  <description>from raw counts single cell RNAseq expression values"</description>
+  <requirements>
+      <requirement type="package" version="2.3.1=r3.4.1_0">r-seurat</requirement>
+      <requirement type="package" version="1.6.0=r3.4.1_0">bioconductor-scater</requirement>
+      <requirement type="package" version="1.0.0=r3.4.1_0">bioconductor-singlecellexperiment</requirement>
+      <requirement type="package" version="1.6.0=r341_0">r-optparse</requirement>
+   </requirements>
+   <stdio>
+      <exit_code range="1:" level="fatal" description="Tool exception" />
+   </stdio>
+   <command detect_errors="exit_code"><![CDATA[
+
+Rscript $__tool_directory__/filter_cells_and_genes.R
+    -f $file
+    -c $file_header
+    -s $file_sep
+    -m $option.is_mito
+    #if $option.is_mito == "TRUE":
+        --header_mito $option.header_mito
+    #end if
+    -p $method.package
+    #if $method.package == "scater":
+        -g $method.gene_filter
+        --min_cells $method.min_cells
+    #end if
+    #if $method.package == "Seurat":
+        --min_genes $method.min_genes
+        --min_cells $method.min_cells
+    #end if
+    --output_pdf $output_plot
+    --output_matrix $output_matrix
+
+]]></command>
+   <inputs>
+      <param name="file" type="data" format="txt" label="Raw counts of expression data"/>
+      <param name="file_sep" type="select" label="Input column separator">
+         <option value="tab" selected="true">Tabs</option>
+         <option value=",">Commas</option>
+         <option value=" ">Whitespaces</option>
+         <option value=".">Dots</option>
+         <option value="_">Underscores</option>
+         <option value="-">Dashes</option>
+   </param>
+   <param name="file_header" type="select" label="Consider first line of input file as header?">
+      <option value="TRUE" selected="true">Yes</option>
+      <option value="FALSE">No</option>
+   </param>
+   <conditional name="option">
+      <param name="is_mito" type="select" label="Are there mitochondrial genes in the expression data ? Do you want to remove cells that detect more than 20% of mitochondrial genes ?">
```

Comment:
2 questions in a labels is one in excess ! 

```
label="Do you want to remove cells that detect more than 20% of mitochondrial genes ? (only applicable if mitochondrial genes are in the expression data)"
```

Moreover, this has to be tested, naturally.